mms/protocol/otf_message_handler.py

```diff
@@ -0,0 +1,241 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+OTF Codec
+"""
+import struct
+import json
+
+from mms.mxnet_model_service_error import MMSError
+from mms.utils.model_server_error_codes import ModelServerErrorCodes as Err
+
+int_size = 4
+double_size = 8
+END_OF_LIST = -2
+START_OF_LIST = -1
+
+
+class OtfCodecHandler(object):
+    """
+    OTF Codec class
+    """
+
+    def _retrieve_buffer(self, conn, length):
+        data = bytearray()
+
+        try:
+            while length > 0:
+                pkt = conn.recv(length)
+                data += pkt
+                length -= len(pkt)
+        except (IOError, OSError) as sock_err:
```

Comment:
Convert native exception to MMSError doesn't provide any value

Reply:
It does. I look for the MMS error in codec to exit the worker.